argocd_agent_installation/argocd-agent-installation.adoc

```diff
@@ -0,0 +1,72 @@
+
+For an architectural overview and more details about Argo CD Agent, see the `Understanding Argo CD Agent architecture` section.
+
+[id="prerequisites_{context}"]
+== Prerequisites
+
+* You have installed the {gitops-title} Operator on two {OCP} clusters.
+* You have installed the OpenShift CLI (`oc`).
+* You have installed a `cluster-scoped Argo CD` instance on the hub cluster.
+* You have configured `Apps in any namespace` feature on the hub cluster.
+* You have installed `argocd-agentctl` CLI tool.
+* You have installed `helm` CLI for your Agent setup. Ensure that the `helm` CLI version is greater than v3.8.0.
+* You have read the key Argo CD terms required to understand the Principal and Agent components in the _Argo CD Terminologies_ section.
+
+// Argo CD Agent Terminologies
+include::modules/gitops-argocd-agent-terminologies.adoc[leveloffset=+1]
+
+// Installing the Principal component
+include::modules/gitops-argocd-agent-install-principal-component.adoc[leveloffset=+1]
+
+// Setting up a spoke cluster environment
+include::modules/gitops-setting-up-a-spoke-cluster-environment.adoc[leveloffset=+1]
+
+// Installing the Argo CD Agent in managed mode by using a Helm chart
+include::modules/gitops-installing-argocd-agent-in-managed-mode-by-using-helm-chart.adoc[leveloffset=+1]
+
+// Installing the Argo CD Agent in autonomous mode by using a Helm chart
+include::modules/gitops-installing-argocd-agent-in-autonomous-mode-by-using-helm-chart.adoc[leveloffset=+1]
+
+// Verifying the Argo CD Agent installation in Managed Mode or Autonomous Mode
+include::modules/gitops-verifying-argocd-agent-installation-in-managed-or-autonomous-mode.adoc[leveloffset=+1]
+
+// Deploying Argo CD applications
+include::modules/gitops-deploying-argocd-applications.adoc[leveloffset=+1]
+
+// Deploying an Argo CD application in managed mode
+include::modules/gitops-deploy-argocd-application-in-managed-mode.adoc[leveloffset=+2]
+
+// Deploying an Argo CD application in autonomous mode
+include::modules/gitops-deploy-argocd-application-in-autonomous-mode.adoc[leveloffset=+2]
+
+// Troubleshooting Principal-Agent communication and deployment issues
+include::modules/gitops-troubleshooting-argocd-principal-agent.adoc[leveloffset=+1]
+
+[role="_additional-resources"]
+[id="additional-resources_{context}"]
+== Additional resources
+
+* xref:../argocd_applications/managing-apps-in-non-control-plane-namespaces.adoc[Apps in any namespaces]
+
+* xref:../declarative_clusterconfig/configuring-an-openshift-cluster-by-deploying-an-application-with-cluster-configurations.adoc#using-argo-cd-instance-to-manage-cluster-scoped-resources_configuring-an-openshift-cluster-by-deploying-an-application-with-cluster-configurations[cluster-scoped Argo CD instance]
+
+* link:https://github.com/argoproj-labs/argocd-agent/releases[argocd-agentctl CLI tool (Argo CD Agent upstream GitHub)]
```

Comment:
The downstream binary builds for agentctl will be published here: https://developers.redhat.com/content-gateway/rest/browse/pub/cgw/openshift-gitops/

Should we point user to this location? @jparsai @jgwest

Reply:
@Dhruv-Soni11 - can we update the link?
```suggestion
* link:https://developers.redhat.com/content-gateway/rest/browse/pub/cgw/openshift-gitops[argocd-agentctl CLI tool]
```

Reply:
Fixed.